schema/dependency.go

```diff
@@ -39,8 +39,9 @@ func (s Schema) validateDependencies(changes map[string]interface{}, doc map[str
 		path := prefix + name
 		field := s.GetField(path)
 		if field != nil && field.Dependency != nil {
-			if !field.Dependency.Match(doc) {
-				addFieldError(errs, name, fmt.Sprintf("does not match dependency: %+v", field.Dependency))
+			p := field.Dependency(s)
```

Comment:
I guess I need to precache that `Predicate` somewhere, and not parse it every time.

Reply:
If you leave it as an "unprepared" query instead of a func, the `resource.Index.Bind` and `resource.Resource.Bind` methods could potentially prepare and store it. It does a similar form of override where it creates `schema.Connection` fields (the reverse of `schema.Resource`)